caas/Dockerfile

```diff
@@ -4,8 +4,10 @@ ARG TARGETARCH
 ARG BUILDOS
 
 # Add the syslog user for audit logging.
-RUN useradd --system -M syslog
-RUN usermod -s /usr/sbin/nologin syslog
+RUN useradd --system --no-create-home --shell /usr/sbin/nologin syslog
+# Add the juju user for rootless agents
+RUN groupadd --gid 170 juju
```

Comment:
Can we add a comment in both locations saying the default in Juju needs to be updated and vice versa? Just a nit.